github/repos.go

```diff
@@ -501,29 +501,56 @@ func (s *RepositoriesService) ListTags(owner string, repo string, opt *ListOptio
 
 // Branch represents a repository branch
 type Branch struct {
-	Name       *string     `json:"name,omitempty"`
-	Commit     *Commit     `json:"commit,omitempty"`
-	Protection *Protection `json:"protection,omitempty"`
+	Name      *string `json:"name,omitempty"`
+	Commit    *Commit `json:"commit,omitempty"`
+	Protected *bool   `json:"protected,omitempty"`
 }
 
-// Protection represents a repository branch's protection
+// Protection represents a repository branch's protection.
 type Protection struct {
-	Enabled              *bool                 `json:"enabled,omitempty"`
-	RequiredStatusChecks *RequiredStatusChecks `json:"required_status_checks,omitempty"`
+	RequiredStatusChecks *RequiredStatusChecks `json:"required_status_checks"`
+	Restrictions         *BranchRestrictions   `json:"restrictions"`
 }
 
-// RequiredStatusChecks represents the protection status of a individual branch
+// ProtectionRequest represents a request to create/edit a branch's protection.
+// It is separate from Protection above because the request structure is
+// different from the response structure.
```

Comment:
I _think_ the entire "It is separate from Protection above because the request structure is different from the response structure." sentence can be left out. It should be pretty self-evident from the code itself.

But I might be biased because I've seen tons of Go code that does this, so it's up to you if you wanna keep it.